source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,242 @@
+====================
+Object Serialization
+====================
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+**Object serialization** is the process of translating an in-memory
+object into a format that can be stored persistently. The reverse process
+is called deserialization. MongoDB stores data as documents containing
+fields with BSON values. The driver uses object
+serialization to translate data between BSON documents and Java objects.
+You can specify how the driver should perform serialization
+using **codecs**.
+
+Codecs implement translation between Extended JSON and Java objects
+according to the ``Codec`` interface using methods that serialize and
+deserialize data. A ``CodecRegistry`` is a class that contains a list
+of codec instances. You can specify a ``CodecRegistry`` for a
+``MongoClient``, ``MongoDatabase``, or ``MongoCollection`` to be
+used for all subsequent operations on that object.
+Whenever you send or receive data from a connected MongoDB instance,
+the driver selects the first codec that can translate each object in the
+``CodecRegistry`` of the ``MongoClient``, ``MongoDatabase``, or
+``MongoCollection`` used to communicate with the MongoDB instance.
+
+.. _default-codecs:
+
+Default Codecs
+--------------
+
+By default, every ``MongoClient`` uses a ``CodecRegistry`` that contains
+codecs allowing the serialization of three classes and many standard
+Java types those classes can contain, including objects of type
+``Integer``, ``Long``, ``Date``, and more:
+
+.. list-table:: Default Serializable Classes
+   :widths: 40 60
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Class
+     - Codec
+
+   * - :java-docs:`Document <apidocs/bson/org/bson/Document.html>`
+     - :java-docs:`DocumentCodec <apidocs/bson/org/bson/codecs/DocumentCodec.html>`
+
+   * - :java-docs:`BasicDBObject <apidocs/mongodb-driver-core/com/mongodb/BasicDBObject.html>`
+     - :java-docs:`DBObjectCodec <apidocs/mongodb-driver-core/com/mongodb/DBObjectCodec.html>`
+
+   * - :java-docs:`BsonDocument <apidocs/bson/org/bson/BsonDocument.html>`
```

Comment:
This is the type safe Bson Document class, where all values directly reflect a Bson type.